src/main/java/com/ixortalk/organization/api/events/OrganizationEventHandler.java

```diff
@@ -67,39 +67,41 @@
     @Inject
     private AssetMgmtFacade assetMgmtFacade;
 
+    @Inject
+    private SecurityService securityService;
+
     @HandleBeforeCreate
     public void handleBeforeCreate(Organization organization) {
-        organization.generateRoleName();
-
         organizationRestResource.findByName(organization.getName()).ifPresent(existing -> {
             throw new ConflictException();
         });
-        organizationRestResource.findByRole(organization.getRole()).ifPresent(existing -> {
-            throw new ConflictException();
-        });
-
-        validateRoleDoesNotExistInAuth0(organization.getRole());
     }
 
     @HandleAfterCreate
     public void handleAfterCreate(Organization organization) {
-        auth0Roles.addRole(organization.getRole());
-        userEmailProvider.getCurrentUsersEmail().ifPresent(email -> auth0Roles.assignRolesToUser(email, newHashSet(organization.getRole())));
+
     }
 
     @HandleBeforeDelete
     public void handleBeforeDelete(Organization organization) {
-        if (!organization.getUsers().isEmpty()) throw new BadRequestException("Users not empty");
+        //only allow deletion when no users available in the database, or when current user is the only admin user of the organization.
```

Comment:
Can this check be refactored to a single check/method "contains only current user"?  And appropriate error message?